spyder_line_profiler/spyder/widgets.py

```diff
@@ -371,48 +483,94 @@ def read_output(self, error=False):
         self.output = self.error_output + self.output
-        # FIXME: figure out if show_data should be called here or
-        #        as a signal from the combobox
-        self.show_data(justanalyzed=True)
+        if not self.output == 'aborted':
+            elapsed = str(datetime.now() - self.started_time).split(".")[0]
+            self.show_data(justanalyzed=True)
+            self.datelabel.setText(_(f'Profiling finished after {elapsed}'))
+        self.show_errorlog()  # If errors occurred, show them.
         self.sig_finished.emit()
-
+        
     def kill_if_running(self):
+        self.datelabel.setText(_('Profiling aborted.'))
         if self.process is not None:
             if self.process.state() == QProcess.Running:
                 self.process.kill()
+                self.output = 'aborted'
                 self.process.waitForFinished()
-
+                
+    def clear_data(self):
+        self.datatree.clear()
+        self.clear_action.setEnabled(False)
+        self.log_action.setEnabled(False)
+        self.save_action.setEnabled(False)
+        self.output = ''
+        
     def show_data(self, justanalyzed=False):
         if not justanalyzed:
-            self.output = None
-        self.log_button.setEnabled(
-            self.output is not None and len(self.output) > 0)
+            self.clear_data()
+        output_exists = self.output is not None and len(self.output) > 0
+        self.clear_action.setEnabled(output_exists)
+        self.log_action.setEnabled(output_exists)
+        self.save_action.setEnabled(output_exists)
+
         self.kill_if_running()
         filename = to_text_string(self.filecombo.currentText())
         if not filename:
             return
 
         self.datatree.load_data(self.DATAPATH)
-        self.datelabel.setText(_('Sorting data, please wait...'))
         QApplication.processEvents()
         self.datatree.show_tree()
 
         text_style = "<span style=\'color: #444444\'><b>%s </b></span>"
         date_text = text_style % time.strftime("%d %b %Y %H:%M",
                                                time.localtime())
         self.datelabel.setText(date_text)
+        
+    def save_data(self):
+        """Save data."""
+        if not self.output:
+            self.datelabel.setText(_("Nothing to save"))
+            return
+        
+        title = _( "Save line profiler result")
+        curr_filename = self.filecombo.currentText()
+        filename, _selfilter = getsavefilename(
+            self,
+            title,
+             f'{curr_filename}_lineprof.txt',
+            _("LineProfiler result") + " (*.txt)",
+        )
 
-
+        if filename:
+            with open(filename, 'w') as f:
+                # for some weird reason, everything is double spaced on Win
+                results = self.output
+                print(results)
+                results  = results.replace('\r', '')
+                print(results.encode())
```

Comment:
Remove `print` statements and extra space:
```suggestion
                results = results.replace('\r', '')
```